### PR TITLE
Updating convolution docs

### PR DESCRIPTION
**Pull request by Shashi456**
Fix: #244

### CODE REVIEW

File: Sources/TensorFlow/Operators/NN.swift [hunk 1]
```diff
@@ -384,15 +384,14 @@ func _vjpAvgPool3D<Scalar: TensorFlowFloatingPoint>(
...
-/// filter, strides, and padding.
+/// Computes a 2-D convolution with the specified input, filter, strides, and padding.
 ///
 /// - Parameters:
 ///   - x: The input.
 ///   - filter: The convolution filter.
 ///   - strides: The strides of the sliding filter for each dimension of the input.
 ///   - padding: The padding for the operation.
-/// - Precondition: `self` must have rank 4.
+/// - Precondition: `input` must have rank `4`.
```
Comment (rxwei) [inline Sources/TensorFlow/Operators/NN.swift]: Oh, there is no parameter called `input`. It's currently called `x`. Could you please help change the first parameter name to `input` as well as the doc comment on line 390?

File: Sources/TensorFlow/Operators/NN.swift [hunk 2]
```diff
@@ -409,15 +408,14 @@ public func conv2D<Scalar: TensorFlowFloatingPoint>(
...
-/// filter, strides, and padding.
+/// Computes a 3-D convolution with the specified input, filter, strides, and padding.
 ///
 /// - Parameters:
 ///   - x: The input.
 ///   - filter: The convolution filter.
 ///   - strides: The strides of the sliding filter for each dimension of the input.
 ///   - padding: The padding for the operation.
-/// - Precondition: `self` must have rank 5.
+/// - Precondition: `input` must have rank `5`.
```
Comment (rxwei) [inline Sources/TensorFlow/Operators/NN.swift]: Same issue here.

Comment (Shashi456) [reply Sources/TensorFlow/Operators/NN.swift]: @rxwei why did we change it to input in #241 then o.O? I'll change it to x there as well. cool.

Comment (rxwei) [reply Sources/TensorFlow/Operators/NN.swift]: I actually suggested changing all the `x`'s to `input`.

Comment (Shashi456) [reply Sources/TensorFlow/Operators/NN.swift]: @rxwei what about the comments in the parameters?

Comment (rxwei) [reply Sources/TensorFlow/Operators/NN.swift]: Yes.

Comment (Shashi456) [reply Sources/TensorFlow/Operators/NN.swift]: @rxwei one final thing, Should I change all the x's in nn.swift to input?

Comment (rxwei) [reply Sources/TensorFlow/Operators/NN.swift]: For convolution and pooling layers, yes.

File: Sources/TensorFlow/Operators/NN.swift [hunk 3]
```diff
@@ -384,50 +384,48 @@ func _vjpAvgPool3D<Scalar: TensorFlowFloatingPoint>(
     })
 }
 
-/// Computes a 2-D convolution using `self` as input, with the specified
-/// filter, strides, and padding.
+/// Computes a 2-D convolution with the specified input, filter, strides, and padding.
 ///
 /// - Parameters:
-///   - x: The input.
+///   - input.
```
Comment (rxwei) [inline Sources/TensorFlow/Operators/NN.swift]: This is not the correct documentation comment syntax. [Swift API Design Guidelines - Fundamentals](https://swift.org/documentation/api-design-guidelines/#fundamentals) which describes the documentation comment syntax.

You should add a colon after the _parameter name_, followed by a _noun phrase_ that describes the parameter, followed by a period.
```suggestion
///   - input: The input.
```

Comment (Shashi456) [reply Sources/TensorFlow/Operators/NN.swift]: @rxwei alright, i just find this very redundant is all. I'll update it accordingly.

**Pull request merged**